SystemsGenetics/GEMmaker
# Cleanup of Downloaded FASTQ files

Resolves issue #67 

## Description
1.  Adds a process to cleanup downloaded SRA run FASTQ files (SRR fastq files)
2.  Adds a process to cleanup merged SRR fastq files (SRX fastq files)
3.  Renames all of the channels for cleaning to be more intuitive.
4.  Adds a new `params.output.publish_downloaded_fastq` option to the nextflow.config.example file.
5.  Because each cleanup process is identical, I moved the code into a new template file which you can find in a new templates directory.

## How to Test
1.  Make sure you copy the new `params.output.publish_downloaded_fastq` param over to your config file.
2.  Enable publishing of `downloaded_fastq`, `trimmed_fastq` and `bam`
3.  Run GEMmaker with both the local and remote example sets together.
4.  Of all the clean processes, you should only see the `cleam_sam` process in your output. This is because we always clean up the sam files.
4.  Now, edit your config file and disable publishing of `downloaded_fastq`, `trimmed_fastq` and `bam`
5. Remove the `output`, `work` and `.nextflow` directories
    ``` rm -rf output work .nextflow```
6. Re-run.  This time you should see all of the clean processes in the output: `clean_downloaded_fastq`, `clean_merged_fastq`, `clean_sam` and `clean_bam`.
7. If you examine the `output` directory you should see no `.fastq` or .`bam` files.
8. If you examine the `work` directory you should see all of the files but the `du` command should report that all `.sam`, `.bam` and `.fastq` files are of size zero.

If all of this works for you then the test was successful and can be merged.
